mmdeploy/codebase/mmdet/models/dense_heads/yolox_head.py

```diff
@@ -91,3 +91,121 @@ def yolox_head__get_bboxes(ctx,
     return multiclass_nms(bboxes, scores, max_output_boxes_per_class,
                           iou_threshold, score_threshold, pre_top_k,
                           keep_top_k)
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmdet.models.dense_heads.YOLOXHead.get_bboxes', backend='ncnn')
+def yolox_head__get_bboxes__ncnn(ctx,
+                                 self,
+                                 cls_scores,
+                                 bbox_preds,
+                                 objectnesses,
+                                 img_metas=None,
+                                 cfg=None,
+                                 rescale=False,
+                                 with_nms=True):
+    """Rewrite `get_bboxes` of YOLOXHead for ncnn backend.
+
+    1. Decode the prior to a box format for ncnn DetectionOutput layer to do
+    the post-processing.
+    2. Batch dimension is not supported by ncnn, but supported by pytorch.
+    The negative value of axis in torch.cat is rewritten as corresponding
+    positive value to avoid axis shift.
+    3. 2-dimension tensor broadcast of `BinaryOps` operator is not supported by
+    ncnn. This function unsqueeze 2-dimension tensor to 3-dimension tensor for
+    correct `BinaryOps` calculation by ncnn.
+
+    Args:
+        ctx: Context that contains original meta information.
+        self: Represent the instance of the original class.
+        cls_scores (list[Tensor]): Classification scores for all
+            scale levels, each is a 4D-tensor, has shape
+            (batch_size, num_priors * num_classes, H, W).
+        bbox_preds (list[Tensor]): Box energies / deltas for all
+            scale levels, each is a 4D-tensor, has shape
+            (batch_size, num_priors * 4, H, W).
+        objectnesses (list[Tensor], Optional): Score factor for
+            all scale level, each is a 4D-tensor, has shape
+            (batch_size, 1, H, W).
+        img_metas (list[dict]): Image meta info. Default None.
+        cfg (mmcv.Config, Optional): Test / postprocessing configuration,
+            if None, test_cfg would be used.  Default None.
+        rescale (bool): If True, return boxes in original image space.
+            Default False.
+        with_nms (bool): If True, do nms before return boxes.
+            Default True.
+
+    Returns:
+        output__ncnn (Tensor): outputs, shape is [N, num_det, 6].
+    """
+    from mmdeploy.codebase.mmdet.core.ops import ncnn_detection_output_forward
+    img_height = int(img_metas[0]['img_shape'][0])
```

Comment:
yolox ncnn does not support dynamic shape. may add warning if input config is dynamic .